tests/DownstreamRestApiTests/DownstreamRestApiTests.cs

```diff
@@ -0,0 +1,132 @@
+using Microsoft.Identity.Abstractions;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using UnitTests;
+using Xunit;
+
+namespace UnitTests
+{
+    public class DownstreamRestApiTests
+    {
+        [Fact]
+        public void CloneClonesAllProperties()
+        {
+            var downstreamRestApiOptions = new DownstreamRestApiOptions
+            {
+                Scopes = new[] { "https://apitocall.domain.com/read" },
+                AcquireTokenOptions = new AcquireTokenOptions
+                {
+                    AuthenticationOptionsName = "AzureAd",
+                    Claims = "claims",
+                    CorrelationId = Guid.NewGuid(),
+                    ExtraHeadersParameters = new Dictionary<string, string> { { "slice", "test" } },
+                    ExtraQueryParameters = new Dictionary<string, string> { { "slice", "test" } },
+                    ForceRefresh = true,
+                    LongRunningWebApiSessionKey = AcquireTokenOptions.LongRunningWebApiSessionKeyAuto,
+                    PopPublicKey = "PopKey",
+                    Tenant = "domain.com",
+                    UserFlow = "susi"
+
+                },
+                BaseUrl = "https://apitocall.domain.com",
+                CustomizeHttpRequestMessage = message => message.Headers.Add("x-sku", "sku-value"),
+                Deserializer = value => value,
+                Serializer = input => (input != null) ? new StringContent(input.ToString()!, new MediaTypeHeaderValue("text/json", "UTF8")) : null,
+                HttpMethod = HttpMethod.Trace,
+                ProtocolScheme = "bearer",
+                RelativePath = "/api/values",
+                RequestAppToken = true
+            };
+
+            Assert.Equal("https://apitocall.domain.com/api/values", downstreamRestApiOptions.GetApiUrl());
+
+            var authorizationHeaderProviderOptions = new AuthorizationHeaderProviderOptions(downstreamRestApiOptions);
+            var authorizationHeaderProviderOptionsClone = authorizationHeaderProviderOptions.Clone();
+
+            var downstreamRestApiClone = downstreamRestApiOptions.Clone();
```

Comment:
With the new option 'clone' below would be of type AuthorizationHeaderProviderOptions
```cs
AuthorizationHeaderProviderOptions authorizationHeaderProviderOptions = new DownstreamRestApiOptions();
var clone = authorizationHeaderOptions.Clone();
```